packages/fxa-content-server/app/tests/spec/models/auth_brokers/oauth-webchannel-v1.js

```diff
@@ -304,41 +304,80 @@ describe('models/auth_brokers/oauth-webchannel-v1', () => {
     });
   });
 
-  [
-    'afterCompleteSignInWithCode',
-    'afterSignIn',
-    'afterSignUpConfirmationPoll',
-  ].forEach((brokerMethod) => {
-    describe(brokerMethod, () => {
-      it('notifies the channel of login with required fields', () => {
-        return broker[brokerMethod](account).then((result) => {
-          assert.isTrue(broker.send.calledTwice);
+  describe('afterSignIn', () => {
+    it('redirects to ConnectAnotherDevice, notifies the channel of login with required fields', () => {
+      return broker
+        .afterSignIn(account)
+        .then((behavior) => {
+          assert.equal(behavior.type, 'connect-another-device');
+
           assert.calledWithExactly(broker.send.getCall(0), LOGIN_MESSAGE, {
             sessionToken: 'abc123',
             email: 'test@email.com',
             uid: 'uid',
             verified: true,
             verifiedCanLinkAccount: false,
           });
+          return new Promise((resolve) => {
+            // Hack to ensure 2nd broker.send assertion is after it occurs
+            setTimeout(resolve, 0);
```

Comment:
I'm sure there's a way to this without the hack but I couldn't sort it out in a reasonable time. It's something to do with the now nested behaviors, the 2nd `broker.send` test assertion was running before the 2nd message was being sent without it. I tried spying on other functions but resorted to this. This code will be obsolete with the sunset content-server epic but LMK if you (the reviewer, hey thanks!) has any ideas.